recipes/tensorflow-lite/all/conanfile.py

```diff
@@ -83,40 +83,46 @@ def requirements(self):
         self.requires("gemmlowp/cci.20210928")
         if self.settings.arch in ("x86", "x86_64"):
             self.requires("intel-neon2sse/cci.20210225")
-        self.requires("ruy/cci.20210622")
+        self.requires("ruy/cci.20220628")
         if self.options.with_xnnpack:
-            self.requires("xnnpack/cci.20211210")
+            self.requires("xnnpack/cci.20220621")
+
+        if self.options.with_xnnpack or self.options.get_safe("with_nnapi", False):
             self.requires("fp16/cci.20210320")
 
+    def build_requirements(self):
+        self.tool_requires("cmake/[^3.16]")
+
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, 14)
 
-        def lazy_lt_semver(v1, v2):
-            lv1 = [int(v) for v in v1.split(".")]
-            lv2 = [int(v) for v in v2.split(".")]
-            min_length = min(len(lv1), len(lv2))
-            return lv1[:min_length] < lv2[:min_length]
-
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
         if not minimum_version:
             self.output.warn(f"{self.name} requires C++14. Your compiler is unknown. Assuming it supports C++14.")
-        elif lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
             raise ConanInvalidConfiguration(f"{self.name} requires C++14, which your compiler does not support.")
-        if self.options.shared:
-            if self.settings.os == "Linux" and not self.options["ruy"].shared:
-                raise ConanInvalidConfiguration(
-                        f"The project {self.name}/{self.version} with shared=True on Linux requires ruy:shared=True")
-
-    def build_requirements(self):
-        self.build_requires("ninja/1.10.2")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    def build(self):
+        files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+
+        # Shared build fails on Windows with
+        # simple_memory_arena_debug_dump.obj : error LNK2005: "void __cdecl tflite::DumpArenaInfo(...) already defined in simple_memory_arena.obj
+        # Resolve the conflict by removing the conflicting
+        # implementation for now.
+        if tools.Version(self.version) >= "2.9.1" and self.settings.os == "Windows" and self.options.shared:
+            os.remove(os.path.join(self.source_folder, self._source_subfolder, "tensorflow/lite/simple_memory_arena_debug_dump.cc"))
```

Comment:
can this be moved to a patch? it's more visible

Reply:
done!